pyramid_learning_journal/pyramid_learning_journal/tests.py

```diff
@@ -1,36 +1,150 @@
+    })
+    config.include("pyramid_learning_journal.models")
+
+    def teardown():
+        testing.tearDown()
+
+    request.addfinalizer(teardown)
+    return config
+
+
+@pytest.fixture
+def db_session(configuration, request):
+    """Create a session for interacting with the test database.
+    This uses the dbsession_factory on the configurator instance
+    to create a new database session. It binds that session to the available
+    engine and returns a new session for every call of the dumm_request object.
+    """
+    SessionFactory = configuration.registry["dbsession_factory"]
+    session = SessionFactory()
+    engine = session.bind
+    Base.metadata.create_all(engine)
+
+    def teardown():
+        session.transaction.rollback()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return session
+
+
+@pytest.fixture
+def dummy_request(db_session):
+    """Instantiate a fake HTTP Request, complete with a database session.
+    This is a function-level fixture, so every new request will have a
+    new database session.
+    """
+    return testing.DummyRequest(dbsession=db_session)
+
+
+def test_model_gets_added(db_session):
+    assert len(db_session.query(Journal).all()) == 0
+    model = Journal(
+        title="title filler",
+        posted_date=datetime.datetime.now(),
+        body="fake body content"
+    )
+    db_session.add(model)
+    assert len(db_session.query(Journal).all()) == 1
+
+
+@pytest.fixture(scope="session")
+def testapp(request):
     """Create a test application to use for functional tests."""
-    from pyramid_learning_journal import main
     from webtest import TestApp
-    app = main({})
-    return TestApp(app)
+    from pyramid_learning_journal import main
+
+    app = main({}, **{"sqlalchemy.url": "postgres:///journal_test_db"})
+    testapp = TestApp(app)
+
+    SessionFactory = app.registry["dbsession_factory"]
+    engine = SessionFactory().bind
+    Base.metadata.create_all(bind=engine)
+
+    def tearDown():
+        Base.metadata.drop_all(bind=engine)
+
+    request.addfinalizer(tearDown)
+
+    return testapp
+
+
+@pytest.fixture
+def fill_the_db(testapp):
+    SessionFactory = testapp.app.registry["dbsession_factory"]
+    with transaction.manager:
+        dbsession = get_tm_session(SessionFactory, transaction.manager)
+        dbsession.add_all(JOURNAL_LIST)
+
+    return dbsession
+
+
+def test_home_route_returns_home_content(dummy_request):
+    """Home view returns a dictionary of values."""
+    from pyramid_learning_journal.views.default import home_view
+    response = home_view(dummy_request)
+    assert isinstance(response, dict)
+
 
+def test_home_view_returns_empty_when_database_empty(dummy_request):
+    """Home view returns nothing when there is no data."""
+    from pyramid_learning_journal.views.default import home_view
+    response = home_view(dummy_request)
+    assert len(response['journals']) == 0
 
```

Comment:
Good job getting passing tests. Now try to expand on them. I'd like to see tests for edge cases and expected failures. Test your error handling.